source.js

```diff
@@ -220,8 +220,11 @@ class Source extends Resource {
 
     this.ready((err) => {
       if (err) { return callback(err) }
+
+
       const { uri } = this
-      const stream = this.createReadStream(opts)
+      const stream = this.uri.includes('file://') ?
```

Comment:
Could we have a null check for `this.uri` in case it isn't available and only a stream can be created?

Reply:
Fo' so - a92388d7402b7dddc389cb611d862c337d2a60b6